test/export/dashboard.test.ts

```diff
@@ -0,0 +1,159 @@
+// (C) 2019 GoodData Corporation
+import "isomorphic-fetch";
+import fetchMock from "fetch-mock";
+import { DashboardExport } from "@gooddata/typings";
+import { DashboardModule } from "../../src/export/dashboard";
+import { ApiResponseError, XhrModule } from "../../src/xhr";
+import {
+    ACCEPTED_REQUEST_STATUS,
+    BAD_REQUEST_STATUS,
+    SUCCESS_REQUEST_STATUS,
+} from "../../src/constants/errors";
+import { mockPollingRequest, mockPollingRequestWithStatus } from "../helpers/polling";
+import { IExportResponse } from "../../src/interfaces";
+
+const dashboardModuleMock = () => new DashboardModule(new XhrModule(fetch, {}));
+
+describe("exportDashboard", () => {
+    const projectId = "testProjectId";
+    const dashboardUri = `/gdc/md/${projectId}/obj/dashboard123`;
+    const endpoint = `/gdc/internal/projects/${projectId}/exportDashboard`;
+    const createdPdf = `/gdc/exporter/result/${projectId}/pdfFileUri`;
+
+    const relativeDateFilter: DashboardExport.FilterContextItem = {
+        dateFilter: {
+            type: "relative",
+            from: -11,
+            to: 11,
+            granularity: "GDC.time.month",
+        },
+    };
+
+    const absoluteDateFilter: DashboardExport.FilterContextItem = {
+        dateFilter: {
+            type: "absolute",
+            from: "2019-08-06",
+            to: "2019-08-08",
+            granularity: "GDC.time.month",
+        },
+    };
+
+    const attributeFilter: DashboardExport.FilterContextItem = {
+        attributeFilter: {
+            displayForm: "/gdc/md/testProjectId/obj/700",
+            negativeSelection: false,
+            attributeElements: ["/gdc/md/testProjectId/obj/750", "/gdc/md/testProjectId/obj/751"],
+        },
+    };
+
+    describe("successful pdf export", () => {
+        beforeEach(() => {
+            const mockTask = (status: number) => ({ status, uri: createdPdf });
+            const finishedTask = mockTask(SUCCESS_REQUEST_STATUS);
+            const runningTask = mockTask(ACCEPTED_REQUEST_STATUS);
+            mockPollingRequest(createdPdf, runningTask, finishedTask);
+
+            fetchMock.post(endpoint, {
+                uri: createdPdf,
+            });
+        });
+
+        afterEach(() => {
+            fetchMock.restore();
+        });
+
+        it("should send POST request with payload", () => {
+            return dashboardModuleMock()
+                .exportDashboardToPdf(
+                    projectId,
+                    dashboardUri,
+                    [absoluteDateFilter, relativeDateFilter, attributeFilter],
+                    { pollStep: 1 },
+                )
+                .then(() => {
```

Comment:
we are trying to use primary async-await approach :) please adjust your code :) it will be more readable :)

Reply:
Fixed